client/components/tinymce/plugins/contact-form/dialog/field-list.jsx

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import EmptyContent from 'components/empty-content';
+import Field from './field';
+
+export default React.createClass( {
+	displayName: 'ContactFormDialogFieldList',
+
+	propTypes: {
+		fields: PropTypes.array.isRequired,
+		onAddNewField: PropTypes.func.isRequired,
+		onUpdateField: PropTypes.func.isRequired,
+		onRemoveField: PropTypes.func.isRequired
+	},
+
+	render() {
+		if ( this.props.fields.length > 0 ) {
+			return (
+				<div className="editor-contact-form-modal-fields">
+					{ this.props.fields.map( ( field, index ) => {
+						const { label, type, options, required } = field;
+						return (
+							<Field
+								key={ index }
+								{ ...{ label, type, options, required } }
+								onRemove={ () => this.props.onRemoveField( index ) }
```

Comment:
We're trigging the eslint rule: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md
For performance reasons we want to avoid bind/fat arrow. See: p4TIVU-2nh-p2

In this case, we should be able to avoid this. Pass through index as a prop to Field, and in Field.jsx add a method like the following:
```javascript
handleRemoveClick( event ) {
     this.props.onRemoveField( event, this.props.index );
}
```
and
```javascript
const remove = <Gridicon icon="trash" onClick={ this.handleRemoveClick } className="editor-contact-form-modal-field__remove" />;
```

Same deal with onUpdate below.